lms/djangoapps/grades/tests/test_signals.py

```diff
@@ -251,3 +256,37 @@ def test_disconnect_manager_bad_arg(self):
         with self.assertRaises(ValueError):
             with disconnect_submissions_signal_receiver(PROBLEM_RAW_SCORE_CHANGED):
                 pass
+
+
+@ddt.ddt
+class UserPartitionChangeHandlerTest(TestCase):
+    """
+    Tests to ensure that sending signals related to user
+    partition changes triggers a recalculation of the affected
+    user's course grade.
+    """
+    def setUp(self):
+        super(UserPartitionChangeHandlerTest, self).setUp()
+        self.user = UserFactory()
+        self.course_key = CourseLocator("test_org", "test_course_num", "test_run")
+
+    @patch('lms.djangoapps.grades.signals.handlers.CourseGradeFactory.update')
+    @patch('lms.djangoapps.grades.signals.handlers.CourseGradeFactory.read')
+    @ddt.data(*itertools.product((COHORT_MEMBERSHIP_UPDATED, ENROLLMENT_TRACK_UPDATED), (True, False), (True, False)))
+    @ddt.unpack
+    def test_enrollment_signal(self, signal, write_only_if_engaged, has_grade, read_mock, update_mock):
+        """
+        Tests the grades handler for user partitions-related signals.
+        The handler should call CourseGradeFactory().update with the
+        args below, *except* if the WRITE_ONLY_IF_ENGAGED waffle flag
+        is inactive and the user does not have a grade.
+        """
+        if has_grade:
+            read_mock.return_value = MagicMock()
```

Comment:
I'm curious - why do you need `read_mock`'s return value to be a MagicMock?

Reply:
A mocked function's return value will be a `mock.Mock()` by default, and Mock objects are truthy.  I think what you want is:
```python
if not has_grade:
    read_mock.return_value = None
```